asapdiscovery/data/pdb.py

```diff
@@ -30,39 +29,51 @@ def download_pdb_structure(pdb_id: str, directory: str):
         The PDB ID of interest.
     directory: str or Path, default=user_cache_dir
         The directory for saving the downloaded structure.
+    file_format: str, optional
+        Indicates whether you would like to download the entry in pdb ("pdb") or cif format ("cif"),
+        or the first biological assembly in cif format ("cif1"). Defaults to "pdb".
 
     Returns
     -------
-    : Path or False
-        The path to the the downloaded file if successful, else False.
+    file_path : Path or False
+        The path to the downloaded file if successful, else False.
     """
     from .utils import download_file
     import os
 
-    # check for structure in PDB format
-    url = f"https://files.rcsb.org/download/{pdb_id}.pdb"
-    pdb_path = os.path.join(directory, f"rcsb_{pdb_id.upper()}.pdb")
-    if not os.path.exists(pdb_path):
-        print("Downloading PDB entry in PDB format ...")
-        if download_file(url, pdb_path):
-            return pdb_path
-    else:
-        return pdb_path
-
-    # check for structure in CIF format
-    url = f"https://files.rcsb.org/download/{pdb_id}.cif"
-    cif_path = os.path.join(directory, f"rcsb_{pdb_id.upper()}.cif")
-    if not os.path.exists(cif_path):
-        print("Downloading PDB entry in CIF format ...")
-        if download_file(url, cif_path):
-            return cif_path
+    url_base_str = f"https://files.rcsb.org/download/"  # base str to use for URLs
+    # Dictionary with allowed formats and their upstream basenames
+    format_to_basename = {
+        "pdb": f"{pdb_id.lower()}.pdb",
+        "cif": f"{pdb_id}.cif",
+        "cif1": f"{pdb_id.lower()}-assembly1.cif"
+    }
+
+    allowed_types = format_to_basename.keys()
+    # Make sure pdb_type can be handled
+    if file_format not in allowed_types:
+        raise NotImplementedError(
+            f"pdb_type expected to be one of {allowed_types}, not '{file_format}'"
+        )
+
+    basename = format_to_basename[file_format]
+    local_path = os.path.join(directory, f"rcsb_{basename}")
+    # Download only if it doesn't exist locally
+    if not os.path.exists(local_path):
+        url = f"{url_base_str}{basename}"
+        if download_file(url, local_path):
+            result = local_path
+        else:
+            result = False  # I'm not fond on things returning bools
```

Comment:
would it be better to return `None` or an empty string? or just to raise an exception or something?

Reply:
It would be better to raise an exception whenever the file isn't found, but then it seems like a "major" change of behavior, that's why I didn't propose it in these changes. Since I don't know how much of the rest of the code that uses this function depends on it returning `False`

Reply:
This is kinda nice because if you wanted to download 100 pdb files, and it only downloads 80 for some reason, I think you'd prefer it to go ahead and download 80 and just return a list of the ones it didn't work for rather than dying as soon as one isn't downloaded.

Reply:
@apayne97 when you want that what you do is catching the exception and handle it however you want, that's why it's advisable since you can actually track if a specific exception was raised and do something with it.

Reply:
I think I missed the context of this function, for some reason I thought it was handling a list of URLs, in which case I think throwing an exception in the middle of the list might interrupt pipelines. but since this is only called for a single file at a time, I think it makes sense to raise an exception

Reply:
Yes, I prefer having "atomic" building blocks that do less things than having functions that do a lot, that way you can mix and match in better ways, that also helps testing the API more easily. As far as I can see, this function should be handling downloading just a single file in the format specified. We can make other helper functions that use this to download many files, if needed.

Reply:
yeah I fully agree. is this something that we want to implement in this PR? I'd be happy to take care of it if so

Reply:
Yes, I think we should be able to work this one out in this same PR, it shouldn't take much effort. Please consider then trying to catch the exception in the `download_PDBs` function. Thanks!

Reply:
@ijpulidos I think I've implemented what you wanted but please take a look when you get the chance